mongo_connector/dest_mapping.py

```diff
@@ -162,8 +242,49 @@ def map_db(self, plain_src_db):
         databases should exist and already been put to plain_db when doing
         create/insert operation.
         """
+        if not self.regex_map and not self.plain:
+            return [plain_src_db]
+        # Lookup this namespace to seed the plain_db dictionary
+        self.get(plain_src_db + '.$cmd')
+        return list(self.plain_db.get(plain_src_db, set()))
 
-        if self.plain_db:
-            return list(self.plain_db.get(plain_src_db, set([])))
+    def fields(self, plain_src_ns):
+        """Get the fields to include and exclude for a given namespace."""
+        mapped = self.get(plain_src_ns)
+        if mapped:
+            return mapped.include_fields, mapped.exclude_fields
         else:
-            return [plain_src_db]
+            return None, None
+
+    def projection(self, plain_src_name, projection):
```

Comment:
Moving exclude_fields/include_fields/projection from OplogThread into DestMapping should probably be part of the next PR for filtering per namespace. The interaction of these fields here is kind of awkward -- I've already cleaned it up in the next change just forgot to remove it.